src/test/java/htsjdk/samtools/cram/structure/ContainerTest.java

```diff
@@ -150,4 +150,47 @@ public void getSpansTest(final List<CramCompressionRecord> records,
         Assert.assertEquals(spanMap.get(expectedReferenceContext), expectedAlignmentSpan);
     }
 
+    // show that we can populate all of the slice indexing fields from the
+    // values in the container's header
+
+    // this is part of the deserialization process, and supports index creation
+
+    @Test
+    public static void sliceOffsetsAndSizesTest() {
+        // this container starts 100,000 bytes into the CRAM stream
+        final int containerStreamByteOffset = 100000;
+
+        // this Container consists of:
+        // a header of size 1234 bytes
+        // a Slice of size 789 bytes
+        // a Slice of size 5555 bytes
```

Comment:
This needs to also have a test for the most common case we expect to see, which is a `Container` with a single, single-reference slice. And also a test with an unmapped reference context slice.

Reply:
I'll add one for single slice.

Not sure about changing Ref Context because it's irrelevant to this test.